workflows/taxprofiler.nf

```diff
@@ -125,11 +125,49 @@ workflow TAXPROFILER {
         ch_shortreads_filtered = ch_shortreads_preprocessed
     }
 
+    /*
+        STEP: Run merging
+    */
+
+    if ( params.run_merging ) {
+
+        ch_reads_for_cat_branch = ch_shortreads_filtered
+            .mix( ch_longreads_preprocessed )
+            .map {
+                meta, reads ->
+                    def meta_new = meta.clone()
+                    meta_new.remove('run_accession')
+                    [ meta_new, reads ]
+            }
+            .groupTuple(by: 0)
```

Comment:
Since index zero is the default, I don't think it's necessary.

```suggestion
            .groupTuple()
```

Reply:
When I didn't specify this, this resulted in:

```
[ [meta], [library_1_R1,library_1_R2],  [library_2_R1,library_2_R2] ]
```

Adding this grouped as expected:

```
[ [meta], [library_1_R1,library_1_R2, library_2_R1,library_2_R2] ]
```

Which is what is expected by CAT_FASTQ

Reply:
Errh, what? 😆 That is absolutely not supposed to happen, I think.

Reply:
They're identical as far as I can tell.

```groovy

workflow {
    ch_input = Channel.of(
        [[id: 't1'], [file('lib_1_R1.fastq'), file('lib_1_R2.fastq')]],
        [[id: 't1'], [file('lib_2_R1.fastq'), file('lib_2_R2.fastq')]],
    )

    ch_input.view()

    ch_input.groupTuple().view()

    ch_input.groupTuple(by: 0).view()
}
```

Reply:
THAT'S WHAT I THOUGHT!?

But ok I'll try again tomorrow... maybe it was osme other issue that caused that behaviour I fixed in an another way

Reply:
Ok, looks fine. I think the flatten in two steps down solved that issue

```
[DUMP: run_merging_post_group] [['id':'ERR3201952', 'instrument_platform':'OXFORD_NANOPORE', 'single_end':true], [[/nf-core/test-datasets/taxprofiler/data/fastq/ERR3201952.fastq.gz]]]
[DUMP: run_merging_post_group] [['id':'2612', 'instrument_platform':'ILLUMINA', 'single_end':true], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/6c/2e5ae3b5399bc0611f41d11228c312/2612-ERR5766180.fastq.gz]]
[DUMP: run_merging_post_group] [['id':'2612', 'instrument_platform':'ILLUMINA', 'single_end':false], [[/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/1a/e8a7c96ecf9e0ee21b36e7b9fd14d5/2612-ERR5766176_B_1.fastq.gz, /home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/1a/e8a7c96ecf9e0ee21b36e7b9fd14d5/2612-ERR5766176_B_2.fastq.gz], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/47/71edb0e12cb1fcc1dea1fec5a7920a/2612-ERR5766176_1.fastq.gz, /home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/47/71edb0e12cb1fcc1dea1fec5a7920a/2612-ERR5766176_2.fastq.gz]]]
[DUMP: run_merging_post_group] [['id':'2613', 'instrument_platform':'ILLUMINA', 'single_end':false], [[/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/f4/3358582e854bd4c62d99e8f7120c02/2613-ERR5766181_1.fastq.gz, /home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/f4/3358582e854bd4c62d99e8f7120c02/2613-ERR5766181_2.fastq.gz]]]
[DUMP: run_merging_post_flatten] [['id':'ERR3201952', 'instrument_platform':'OXFORD_NANOPORE', 'single_end':true], [/nf-core/test-datasets/taxprofiler/data/fastq/ERR3201952.fastq.gz]]
[DUMP: run_merging_post_flatten] [['id':'2612', 'instrument_platform':'ILLUMINA', 'single_end':true], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/6c/2e5ae3b5399bc0611f41d11228c312/2612-ERR5766180.fastq.gz]]
[DUMP: run_merging_post_flatten] [['id':'2612', 'instrument_platform':'ILLUMINA', 'single_end':false], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/1a/e8a7c96ecf9e0ee21b36e7b9fd14d5/2612-ERR5766176_B_1.fastq.gz, /home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/1a/e8a7c96ecf9e0ee21b36e7b9fd14d5/2612-ERR5766176_B_2.fastq.gz, /home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/47/71edb0e12cb1fcc1dea1fec5a7920a/2612-ERR5766176_1.fastq.gz, /home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/47/71edb0e12cb1fcc1dea1fec5a7920a/2612-ERR5766176_2.fastq.gz]]
[DUMP: run_merging_post_flatten] [['id':'2613', 'instrument_platform':'ILLUMINA', 'single_end':false], [/home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/f4/3358582e854bd4c62d99e8f7120c02/2613-ERR5766181_1.fastq.gz, /home/jfellows/Documents/git/nf-core/taxprofiler/testing/work/f4/3358582e854bd4c62d99e8f7120c02/2613-ERR5766181_2.fastq.gz]]
```